packages/react-next/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -50,136 +51,353 @@ const OFF_SCREEN_STYLE = { opacity: 0, filter: 'opacity(0)' };
 // calloutMain (the Popup component within the callout)
 const ARIA_ROLE_ATTRIBUTES = ['role', 'aria-roledescription'];
 
-export interface ICalloutState {
-  positions?: ICalloutPositionedInfo;
-  slideDirectionalClassName?: string;
-  calloutElementRect?: ClientRect;
-  heightOffset?: number;
+const DEFAULT_PROPS = {
+  preventDismissOnLostFocus: false,
+  preventDismissOnScroll: false,
+  preventDismissOnResize: false,
+  isBeakVisible: true,
+  beakWidth: 16,
+  gapSpace: 0,
+  minPagePadding: 8,
+  directionalHint: DirectionalHint.bottomAutoEdge,
+} as const;
+
+function useTargets({ target }: ICalloutProps, calloutElement: React.RefObject<HTMLDivElement | null>) {
```

Comment:
pretty neat hook! it would be nice to add some jsdoc comments above; also curious if it could just take in the target - maybe this is generalized enough to be used whenever we want to point something at an element (whether it be an element, string, or ref.)

Reply:
trying not to ask too much here :) good job though.

Reply:
Same with the other hooks, just a doc comment above them would help with understanding purpose and readability.